content/api/commands/getallcookies.md

```diff
@@ -0,0 +1,105 @@
+---
+title: getAllCookies
+---
+
+Get all browser cookies.
+
+## Syntax
+
+```javascript
+cy.getAllCookies()
+cy.getAllCookies(options)
+```
+
+### Usage
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```javascript
+cy.getAllCookies() // Get all cookies
+```
+
+### Arguments
+
+**<Icon name="angle-right"></Icon> options** **_(Object)_**
+
+Pass in an options object to change the default behavior of
+`cy.getAllCookies()`.
+
+| Option    | Default                                                        | Description                                                                              |
+| --------- | -------------------------------------------------------------- | ---------------------------------------------------------------------------------------- |
+| `log`     | `true`                                                         | Displays the command in the [Command log](/guides/core-concepts/cypress-app#Command-Log) |
+| `timeout` | [`responseTimeout`](/guides/references/configuration#Timeouts) | Time to wait for `cy.getAllCookies()` to resolve before [timing out](#Timeouts)          |
+
+### Yields [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Subject-Management)
+
+`cy.getAllCookies()` yields an array of cookie objects. Each cookie object has
+the following properties:
+
+- `domain`
+- `expiry` _(if specified)_
+- `httpOnly`
+- `name`
+- `path`
+- `sameSite` _(if specified)_
+- `secure`
+- `value`
+
+`cy.getAllCookies()` is not a query. It will not update the returned list if
+further cookies are added after it initially executes.
+
+## Examples
+
+### Get Cookies
+
+#### Get cookies after logging in
+
+In this example, we log in through an identity provider which sets a cookie and
+redirects back to our site, which sets a session cookie.
+
+```javascript
+cy.contains('Log in').click()
+cy.origin('https://login.site.com', () => {
+  cy.get('[type=password]').type('*****')
+  cy.contains('Log in').click()
+})
+cy.url().should('include', 'profile')
+cy.getAllCookies()
+  .should('have.length', 2)
+  .then((cookies) => {
+    expect(cookies[0]).to.have.property('name', 'identity_session_id')
+    expect(cookies[1]).to.have.property('name', 'session_id')
+  })
+```
+
+## Rules
+
+### Requirements [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Chains-of-Commands)
+
+- `cy.getAllCookies()` requires being chained off of `cy`.
+
+### Assertions [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Assertions)
+
+- `cy.getAllCookies()` will only run assertions you have chained once, and will
+  not [retry](/guides/core-concepts/retry-ability).
+
+### Timeouts [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Timeouts)
+
+- `cy.getAllCookies()` should never time out.
```

Comment:
If it shouldn't time out why does it have a timeout option?

Reply:
Keep reading 😄 https://github.com/cypress-io/cypress-documentation/pull/4918/files/56aae904c77e5838e6fb63f8f3ca54f8b56d7fcb#diff-09d6a7d88442e5ab446abff108657151514e152d29dbb81e799be35e3993b6c8R90-R96

Reply:
_should_ not _can't_

Reply:
yeah just seems weird it can be configured it it shouldn't 😋

Reply:
That's fair. I guess it's more of safety valve in case there _is_ some issue with automation taking too long even though that's unlikely.